### PR TITLE
Add support for Windows

### LINKED ISSUE #8: Implement Windows trust stores with CryptoAPI
https://docs.microsoft.com/en-us/windows/win32/seccrypto/cryptography-functions#certificate-trust-list-functions

### PR DESCRIPTION
**Pull request by sethmlarson**
Closes #8

### CODE REVIEW

File: truststore.py [hunk 1]
```diff
@@ -380,7 +381,52 @@ def _verify_peercerts_impl(cert_chain: List[bytes], server_hostname=None) -> Non
...
+    WTD_UI_NONE = WINTRUST.WTD_UI_NONE
+    WTD_REVOCATION_CHECK_CHAIN = WINTRUST.WTD_REVOCATION_CHECK_CHAIN
+    WTD_CHOICE_CERT = WINTRUST.WTD_CHOICE_CERT
+    AUTHTYPE_SERVER = WINTRUST.AUTHTYPE_SERVER
+    X509_ASN_ENCODING = WINTRUST.X509_ASN_ENCODING
+    PKCS_7_ASN_ENCODING = WINTRUST.PKCS_7_ASN_ENCODING
+
+    WinVerifyTrust = WINTRUST.WinVerifyTrust
+    WinVerifyTrust.argtypes = [HWND, c_void_p, LPVOID]
+    WinVerifyTrust.restype = LONG
```
Comment (davisagli) [inline truststore.py:415]: https://docs.microsoft.com/en-us/windows/win32/api/wintrust/nf-wintrust-winverifytrust says "For certificate verification, use the [CertGetCertificateChain](https://docs.microsoft.com/en-us/windows/desktop/api/wincrypt/nf-wincrypt-certgetcertificatechain) and [CertVerifyCertificateChainPolicy](https://docs.microsoft.com/en-us/windows/desktop/api/wincrypt/nf-wincrypt-certverifycertificatechainpolicy) functions," which makes me 2nd-guess whether WinVerifyTrust is the right API to use.

Comment (davisagli) [reply truststore.py:415]: It looks like possibly what we should do is:
    - Create an in-memory certificate store using [CertOpenStore](https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certopenstore) with lpszStoreProvider set to CERT_STORE_PROV_MEMORY
    - Add any intermediate certs from our peer to it using [CertAddEncodedCertificateToStore](https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certaddencodedcertificatetostore)
    - Get a cert chain using [CertGetCertificateChain](CertGetCertificateChain) with hAdditionalStore set to the in-memory store
    - Verify the chain using [CertVerifyCertificateChainPolicy](https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certverifycertificatechainpolicy)

**Comment by sethmlarson** [comment]
@davisagli Did you want to implement the Windows certificate verification? I don't have access to a Windows machine and don't know Windows crypto libraries as well as macOS.

**Comment by davisagli** [comment]
@sethmlarson I'm in a similar position -- my comment above is from reading I was doing this morning, not from actual Windows experience. I'll probably continue to poke at it if your attention is elsewhere, but I'm definitely in impostor-syndrome territory and not ready to commit to getting it done in any particular timeframe, and I won't be offended if someone else gets to it first.

**Pull request merged**